microshift.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+source tools.sh
+source snc-library.sh
+
+BUNDLE_TYPE="microshift"
+INSTALL_DIR=crc-tmp-install-data
+CRC_VM_NAME=${CRC_VM_NAME:-crc}
+BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp}
+OPENSHIFT_RELEASE_VERSION=4.12.0
+
+if grep -q -i "release 8" /etc/redhat-release
+then
+  echo "Check if system is registered"
+  # Check the subscription status and register if necessary
+  if ! sudo subscription-manager status >& /dev/null ; then
+     sudo subscription-manager register
+  fi
+else
+  echo "This script only works for RHEL-8"
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+sudo virsh destroy ${CRC_VM_NAME} || true
+sudo virsh undefine --nvram ${CRC_VM_NAME} || true
+sudo rm -fr /var/lib/libvirt/images/${CRC_VM_NAME}.qcow2
+sudo rm -fr /var/lib/libvirt/images/microshift-installer-*.iso
+
+# Download the oc binary for specific OS environment
+OC=./openshift-clients/linux/oc
+download_oc
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+function configure_host {
+    sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
+      cockpit-composer cockpit-machines bash-completion podman genisoimage \
+      createrepo yum-utils selinux-policy-devel jq wget lorax
```

Comment:
Can this list be limited to what the script will use directly ? I'm guessing most of these packages will be downloaded as dependencies of other packages (`bash-completion` for example). If this is c&p'ed from some doc, please add the doc URL in some comment.

Reply:
This is taken from the configure host script which is part of microshift repo. I can reference it as comment. https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh , this script also contain the mock package which we dont' need since we are going to use built RPMs.

Reply:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/composing_a_customized_rhel_system_image/installing-composer_composing-a-customized-rhel-system-image which you mention in the comment only has `yum install osbuild-composer composer-cli cockpit-composer bash-completion` which is a lot more reasonable imo.
The microshift documentation then has `sudo yum install -y yum-utils`
The script uses `git` directly, `git-core` is probably good enough though.

Reply:
Still needs to be cleaned up imo, `sudo dnf install osbuild-composer composer-cli cockpit-composer bash-completion yum-utils lorax git-core  jq` and may `podman`. Some of the packages you listed seem redundant.